### PR TITLE
Add html and man pages for available options

### PR DESCRIPTION
**Pull request by Gerschtli**
Adds rendered list of available options. We could potentially thinks about moving github wiki, readme and changelog documentation to these html pages. 

For local testing, run 
```sh
nix build ".#manualHtml"
open result/share/doc/nix-on-droid/index.html

nix build ".#manPages"
man --local-file result/share/man/man5/nix-on-droid-options.5.gz
```

This PR contains also a new github workflow to automatically deploy latest docs to gh pages on every master push.

Fixed some option descriptions for correct formatting and renamed `Nix-on-Droid` occurrences to `nix-on-droid` for consistency.

Interesting, potential follow-up tasks:
- man page for nix-on-droid
- a `documentation.manPages.enable` option to install the man page on device.

Requires https://github.com/t184256/nix-on-droid/pull/204.
Solves https://github.com/t184256/nix-on-droid/issues/36.

### CODE REVIEW

**Comment by Gerschtli** [comment]
Force push was only rebase on master

**Comment by Gerschtli** [comment]
Oh cool, there is a "Compare"  button to see what changed between force-pushes. Not sure if I never saw it or it was added recently, but then we can just autosquash before pushing and still be able to see the differences :)

File: docs/man-pages.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<!-- Copyright (c) 2019-2022, see AUTHORS. Licensed under MIT License, see LICENSE. -->
+<reference xmlns="http://docbook.org/ns/docbook"
+           xmlns:xlink="http://www.w3.org/1999/xlink"
+           xmlns:xi="http://www.w3.org/2001/XInclude">
+ <title>Nix-on-Droid Reference Pages</title>
```
Comment (t184256) [inline docs/man-pages.xml]: is it going anywhere?

**Review by t184256 — COMMENTED** [review]
Casing: should Android app also lowercase it now? IMO this would look rather odd on Android. (suggestion: lowercase when monospace, heh)

Extra entry:
```
       _module.args

           Type: lazy attribute set of raw value

           Declared by:
               <nix-on-droid/lib/modules.nix>
```
can this be hidden?

**Comment by Gerschtli** [comment]
> Casing: should Android app also lowercase it now? IMO this would look rather odd on Android. (suggestion: lowercase when monospace, heh)

Good point. We can try to do that :D

> _module.args

I tried to hide it, didn't manage to do so. Basically, it is an option you can use, so it is not completely wrong :D Also home-manager docs has (obviously, because of nmd) the same behaviour.

File: docs/man-pages.xml (hunk 1 above)
Comment (Gerschtli) [reply docs/man-pages.xml]: But as mentioned in the PR description
> a `documentation.manPages.enable` option to install the man page on device.

We could install it on the device for easy access to option docs.

**Comment by Gerschtli** [comment]
Tried to find a consistent way for the casing, what do you think?

**Review by t184256 — APPROVED** [review]

**Pull request merged**